lib/std/crypto/pbkdf2.zig

```diff
@@ -52,43 +52,38 @@ const Error = std.crypto.Error;
 ///
 /// rounds: Iteration count. Must be greater than 0. Common values range from 1,000 to 100,000.
 ///         Larger iteration counts improve security by increasing the time required to compute
-///         the derivedKey. It is common to tune this parameter to achieve approximately 100ms.
+///         the dk. It is common to tune this parameter to achieve approximately 100ms.
 ///
 /// Prf: Pseudo-random function to use. A common choice is `std.crypto.auth.hmac.HmacSha256`.
-pub fn pbkdf2(derivedKey: []u8, password: []const u8, salt: []const u8, rounds: u32, comptime Prf: type) Error!void {
+pub fn pbkdf2(dk: []u8, password: []const u8, salt: []const u8, rounds: u32, comptime Prf: type) Error!void {
     if (rounds < 1) return error.WeakParameters;
 
-    const dkLen = derivedKey.len;
-    const hLen = Prf.mac_length;
-    comptime std.debug.assert(hLen >= 1);
+    const dk_len = dk.len;
+    const h_len = Prf.mac_length;
+    comptime std.debug.assert(h_len >= 1);
 
     // FromSpec:
     //
-    //   1. If dkLen > maxInt(u32) * hLen, output "derived key too long" and
+    //   1. If dk_len > maxInt(u32) * h_len, output "derived key too long" and
     //      stop.
     //
-    if (comptime (maxInt(usize) > maxInt(u32) * hLen) and (dkLen > @as(usize, maxInt(u32) * hLen))) {
-        // If maxInt(usize) is less than `maxInt(u32) * hLen` then dkLen is always inbounds
+    if (dk_len / h_len >= maxInt(u32)) {
+        // Counter starts at 1 and is 32 bit, so if we have to return more blocks, we would overflow
         return error.OutputTooLong;
     }
 
     // FromSpec:
     //
-    //   2. Let l be the number of hLen-long blocks of bytes in the derived key,
+    //   2. Let l be the number of h_len-long blocks of bytes in the derived key,
     //      rounding up, and let r be the number of bytes in the last
     //      block
     //
 
-    // l will not overflow, proof:
-    // let `L(dkLen, hLen) = (dkLen + hLen - 1) / hLen`
-    // then `L^-1(l, hLen) = l*hLen - hLen + 1`
-    // 1) L^-1(maxInt(u32), hLen) <= maxInt(u32)*hLen
-    // 2) maxInt(u32)*hLen - hLen + 1 <= maxInt(u32)*hLen // subtract maxInt(u32)*hLen + 1
-    // 3) -hLen <= -1 // multiply by -1
-    // 4) hLen >= 1
-    const r_ = dkLen % hLen;
-    const l = @intCast(u32, (dkLen / hLen) + @as(u1, if (r_ == 0) 0 else 1)); // original: (dkLen + hLen - 1) / hLen
-    const r = if (r_ == 0) hLen else r_;
+    const blocks_count = (dk_len + h_len - 1) / h_len;
```

Comment:
This may overflow on 32 bit. 
if `dk_len` is `maxInt(32) - 1` and `h_len` is 3
then `maxInt(32) - 1 + 3 - 1` = `maxInt(32) + 1` overflows

Reply:
FYI there's also `std.math.divCeil`

Reply:
`std.math.divCeil()` looks perfect here, thank you @LemonBoy!

Changed to `std.math.divCeil()`.